app/components/amount-input-screen/amount-input-screen-ui.tsx

```diff
@@ -53,13 +56,27 @@ export const AmountInputScreenUI: React.FC<AmountInputScreenUIProps> = ({
             {primaryCurrencySymbol && (
               <Text style={styles.primaryCurrencySymbol}>{primaryCurrencySymbol}</Text>
             )}
-            {primaryCurrencyFormattedAmount ? (
-              <Text style={styles.primaryNumberText}>
-                {primaryCurrencyFormattedAmount}
-              </Text>
-            ) : (
-              <Text style={styles.faintPrimaryNumberText}>0</Text>
-            )}
+            <Input
+              value={primaryCurrencyFormattedAmount}
+              showSoftInputOnFocus={false}
+              onChangeText={(e) => {
+                // remove commas for ease of calculation later on
+                const val = e.replaceAll(",", "")
+                // TODO adjust for currencies that use commas instead of decimals
+
+                // test for string input that can be either numerical or float
+                if (/^\d*\.?\d*$/.test(val)) {
+                  const num = Number(val)
+                  onPaste(`${num}`)
+                }
+              }}
```

Comment:
is this code executed both for the case when we add a key or when we paste a string?

Reply:
I double-checked, and no - it's only when the user changes the input field directly themselves.